doc/vim-go.txt

```diff
@@ -1815,6 +1829,200 @@ The `gohtmltmpl` filetype is automatically set for `*.tmpl` files; the
 `gotexttmpl` is never automatically set and needs to be set manually.
 
 
+==============================================================================
+DEBUGGER                                                            *go-debug*
+
+Vim-go comes with a special "debugger mode". This starts a `dlv` process in
+the background and provides various commands to communicate with it.
+
+This debugger is similar to Visual Studio or Eclipse and has the following
+features:
+
+  * Show stack trace and jumps.
+  * List local variables.
+  * List function arguments.
+  * Expand values of struct or array/slice.
+  * Show balloon on the symbol.
+  * Show output of stdout/stderr.
+  * Toggle breakpoint.
+  * Stack operation continue/next/step out.
+
+This feature requires Vim 8.0.0087 or newer with the |+job| feature. Neovim
+does _not_ work (yet).
+This requires a fairly recent version of the Delve debugger (Nov 28 or newer).
+Use `|:GoUpdateBinaries| to install or update it.
+
+                                                              *go-debug-intro*
+GETTING STARTED WITH THE DEBUGGER~
+
+Use |:GoDebugStart| to start the debugger. The first argument is the package
+name, and any arguments after that will be passed on to the program; for
+example:
+>
+    :GoDebugStart . -someflag value
+<
+This may take few seconds. After the code is compiled you'll see three new
+windows: the stack trace on left side, the variable list on the bottom-left,
+and program output at the bottom.
+
+You can add breakpoints with |:GoDebugBreakpoint| (<F9>) and run your program
+with |:GoDebugContinue| (<F5>).
+
+The program will halt on the breakpoint, at which point you can inspect the
+program state. You can go to the next line with |:GoDebugNext| (<F10>) or step
+in with |:GoDebugStep| (<F11>).
+
+Struct values are displayed as `{...}`, array/slices as `[4]`. Use <CR> on the
+variable name to expand the values.
```

Comment:
This is great, but I didn't know I had to go to the `GODEBUG_VARIABLES` window. Worth adding this little piece. Actually we should document more each window. We just mentioned them briefly above.

Reply:
Okay, I rephrased it a bit and added some more info.